[21.09] Fix workflow invocation name overflow

davelopez opened the pull request:
Fixes #12857

![Screenshot from 2022-01-21 12-33-10](https://user-images.githubusercontent.com/46503462/150520198-a56e7449-c1e0-41db-b667-df5475490541.png)

- Added back the "expand" column to make it more obvious that the row has additional details and to display the full name as a tooltip when it gets truncated as @itisAliRH suggested. The previous expand button was removed in https://github.com/galaxyproject/galaxy/pull/9986. @guerler I hope adding back this column makes sense now :)


## How to test the changes?
- [x] Instructions for manual testing are as follows:
  - Follow instructions on #12857

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

davelopez committed:
Fix workflow invocation name overflow

This will force wrapping the name of the workflow when it overflows

davelopez committed:
Use break-word on overflow-wrap

davelopez committed:
Set table columns widths as suggested in comments

davelopez committed:
Add min-width to table

This will prevent the workflow name column to completely dissapear in small resolutions

davelopez committed:
Add explicit row expand button

- Makes more obvious that the row can be expanded.
- Allows to display the full name in the tooltip when truncated.

mvdbeek commented on client/src/components/Workflow/Invocations.vue:
```diff
@@ -33,11 +34,27 @@
                         class="toggle-invocation-details"
                         v-b-tooltip
-                        title="Show Invocation details"
+                        :title="getWorkflowNameByInstanceId(data.item.workflow_id)"
```
Is the the `toggle-invocation-details` class still right here ? And I guess we should have the same tooltip for the history name, which could also be too large

davelopez replied:
Yes, I left the `toggle-invocation-details` class there (is used in tests) and you can still expand the details if you click the name so the previous mechanism is still working for people that are used to it.
I agree the history name column should behave exactly the same and also overflow with ellipsis too instead of wrapping for consistency. Will do that :+1:

mvdbeek replied:
Sounds good, but maybe you can move the `toggle-invocation-details` class to the new button ? I do want to add more context to the workflow and history cells anyway (filter by workflow / history for instance), but I can do that in dev

davelopez replied:
Ok, I moved the class to the new button but still left the expand row details working on the workflow name, you can remove it on dev if you need it when you add the new stuff.

davelopez committed:
Fix column index in unit tests

After adding a new "first" column for expanding

davelopez committed:
Truncate history name too

davelopez committed:
Fix tooltip position

The tooltip was ignoring the text-overflow when the full name was truncated and was positioned in the center of where the full text would be.
Putting the tooltip into a wrapping div seems to solve the issue.

davelopez committed:
Move `toggle-invocation-details` class to expand column

davelopez committed:
Fix column text overflow after wrapping in div

And slightly enhance switch to history tooltip
